### DISCUSSION TITLE
Differences between download files

### DISCUSSION THREAD

**Discussion by xan2622** [RamonUnch/AltDrag#421] [General]
Hi. 
I have noticed that there are several files available on the download page: https://github.com/RamonUnch/AltSnap/releases/tag/1.60

Does the installer AltSnap1.60-x64-inst.exe provide an "automatic update system" ?

I wonder if I should better pick it rather than the AltSnap1.60bin_x64.zip portable version.

**Comment by RamonUnch** [comment]
It really depends on what you want.
If you use the installer, you will get the uninstall option in the Add/Remove programs, and you will also get a shortcut added to the start menu.

If, when you install AltSnap, you choose the old folder as install path, you will have an update of AltSnap, but the old `AltSnap.ini` file will be kept as is. so your configuration will stay the same. In general there is no harm to have an old ini file as all newer settings will have a default value and having old unused settings does not harm anything.

The zipped version simply contains all needed files, so to uninstall you will simply need to delete the folder.
If you want to Update your AltSnap install using the zipped version, you can simply overwrite all files, except the `AltSnap.ini` file to keep your config.

The only case I would recommend to overwrite your old ini file is if you encounter problems after an upgrade, or maybe if you want to directly edit the ini file, in which case the latest default ini file will contains all possible options with a comment.
If you are using the installer the simplest to update while refreshing the config would be to uninstall/re-install AltSnap.

**Comment by RamonUnch** [comment]
File description for 1.60:

**AltSnap1.60-inst.exe**: 32 bit installer (i386)
**AltSnap1.60-x64-inst.exe**: 64 bit installer (x86_64)
**AltSnap1.60bin.zip**: 32 bit *portable version* (i386)
**AltSnap1.60bin_x64.zip**: 64 bit *portable version* (x86_64)
**AltSnap1.60src.zip**: Source code (redundant copy of my folder in case I forgot to sync some changes to the main GitHub repo)

The installer also support update, the only keys that are written to the registry is for the uninstaller, the only file that is written outside of the AltSnap directory is the `AltSnap.lnk` shortcut inside the Programs folder for the Start menu (in case you need to uninstall manually).

The zipped version can be extracted anywhere and will work exactly the same, than the installer version, except that it will not add the shortcut into the start menu and that no uninstall entry will be added to the Add/Remove programs.

In all cases it is possible that AltSnap writes extra key to the registry if you check the Autostart option. The Uninstaller will clean that if needed, you will have to clean it yourself if you used the zipped version, either by unchecking the option before removing the AltSnap folder or by removing the reg key yourself afterward.

The 32 bit version should be compatible with all Windows versions starting with NT4sp3 (i386 and compatible).
The 64 bit version should be compatible With all 64 bit Windows versions starting from XP64 (x86_64 and compatible).

So the supported system are NT4/2000/XP/Vista/7/8.x/10/11. Supporting old Windows is easy, supporting newer Windows is more complicated and that were most of the work goes into.
